src/transformState.js

```diff
@@ -5,7 +5,36 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        removeProperty(state, action.keysToRemove);
+        break;
+
+      case 'clear':
+        clearObject(state);
+        break;
+
+      default:
+        return state;
```

Comment:
In the default case statement, you should not return state, instead use 'break' as this can cause the function to terminate early.